ElevatorSim/src/elevatorsim/floor/Floor.java

```diff
@@ -13,12 +13,14 @@
 public class Floor {
 	private Integer floorNumber;
 	private FloorButton dirButtons;
-	private HashMap<Direction, List<MessageRequest>> activeRequests; 
```

Comment:
I split this map into two lists since thats how many keys there would have been anyway. You were initializing the map but not the lists that would have populated the values -> causing an NPE in your readRequest and loadPassengers functions.

Reply:
Thanks